src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DocumentState.cs

```diff
@@ -359,15 +376,16 @@ private async Task<(RazorCodeDocument, VersionStamp, VersionStamp)> GetGenerated
                 RazorCodeDocument olderOutput = null;
                 var olderInputVersion = default(VersionStamp);
                 var olderOutputVersion = default(VersionStamp);
-                if (_older?.TaskUnsafe != null)
+                var olderTaskUnsafe = _older?.TaskUnsafe;
```

Comment:
I would super like to understand why the fix for this is more complicated than removing some items at line 436

Reply:
Put another way - what code to do we have that calls `GetGeneratedOutput()` and looks at the syntax tree.

Reply:
In VS - Nothing. 
In VSCode - Everything

I'd rather keep the DocumentSnapshots in a state where they can be used as expected without gaps.

Reply:
That's not really possible if they are stateful.

Reply:
Talked offline, the state this PR is in at the moment makes the `DocumentSnapshot` system act in a way where it may or may not have cached state, if it doesn't it re-calculates it. The model works but we might as well enforce that by default without having to call `ClearStoredState`. Going to revamp this PR to utilize WeakReference's to reference output.